pkg/scheduler/scheduler.go

```diff
@@ -155,6 +170,13 @@ func (s *Scheduler) Run(ctx context.Context) {
 	stopCh := ctx.Done()
 	klog.Infof("Starting karmada scheduler")
 	defer klog.Infof("Shutting down karmada scheduler")
+
+	// Establish all connections first and then begin scheduling.
+	if s.enableSchedulerEstimator {
+		s.establishEstimatorConnections()
```

Comment:
Seems we don't need to establish the connection explicitly?

Reply:
Well, this function helps establish the connection first and then begin scheduling. If not explicitly, the scheduler begins to work but the connection has not been established.

Reply:
In other words, we don't want to schedule workloads when the connections have not been established when setting up.

Reply:
You are right. The scheduler may start working before we set up the connections for all clusters.